cluster/gce/util.sh

```diff
@@ -864,6 +864,13 @@ function construct-windows-kubelet-flags {
   WINDOWS_CONTAINER_RUNTIME_ENDPOINT=${KUBE_WINDOWS_CONTAINER_RUNTIME_ENDPOINT:-npipe:////./pipe/containerd-containerd}
   flags+=" --container-runtime-endpoint=${WINDOWS_CONTAINER_RUNTIME_ENDPOINT}"
 
+  # If ENABLE_AUTH_PROVIDER_GCP is set to true, kubelet is enabled to use out-of-tree auth
+  # credential provider. https://kubernetes.io/docs/tasks/kubelet-credential-provider/kubelet-credential-provider
+  if [[ "${ENABLE_AUTH_PROVIDER_GCP:-false}" == "true" ]]; then
```

Comment:
if we default to false, is the plan to add a new test job that exercise the windows tests with this enabled?

Reply:
I was thinking about using existing test. Wouldn't it be enough to set `ENABLE_AUTH_PROVIDER_GCP` to true in `gce/config-test.sh` ?

Reply:
Created PR enabling image credential provider in tests: https://github.com/kubernetes/kubernetes/pull/117671

Reply:
I've rebased this PR to include changes from https://github.com/kubernetes/kubernetes/pull/117671 and looking at [kubelet logs](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/117522/pull-kubernetes-e2e-gce/1652062403971518464/artifacts/e2e-9b5db9cdbc-674b9-minion-group-6743/kubelet.log) auth-provider-gcp plugin is being registered.

Reply:
Amazing! I believe with https://github.com/kubernetes/kubernetes/pull/117503, it is better to switch to use GCP Credential provider by default for all GCE tests. So thank you for setting the default true value in the config for tests.

Reply:
Actually, reading the comment in here: https://github.com/kubernetes/kubernetes/pull/117671#issuecomment-1528751585, I think you will be blocked from merging your changes since there might be some tests that could fail until the concerned people have filtered/changed other tests that could rely on in-tree provider setting. It might be worth adding a separate job in [sig-windows-gce](https://testgrid.k8s.io/sig-windows-gce) dashboard maybe to begin with? cc: @ibabou

Reply:
This PR is gated by `ENABLE_AUTH_PROVIDER_GCP` so it doesn't change anything by itself.

The braking PR was https://github.com/kubernetes/kubernetes/pull/117671 which changed `ENABLE_AUTH_PROVIDER_GCP` value in tests but it has been already merged. So if anything this pr should fix tests instead of break them as it adds support for arm and windows. I guess it could potentially break amd64 but presubmit `pull-kubernetes-e2e-gce` checks that.

Reply:
ok, let's try landing this and we can revert if needed.